.github/workflows/check-public-apis.yml

```diff
@@ -0,0 +1,76 @@
+name: Check for new public APIs
+on:
+  pull_request:
+    types: [opened, labeled, unlabeled, synchronize]
+    paths:
+      - '**/*.swift'
+      - '!StripeFinancialConnections/**'
+      - '!StripeIdentity/**'
+      - '!StripeConnections/**'
+    branches-ignore:
+      - 'releases/*' # ignore release PRs due to them modifying StripeAPIConfiguration.STPSDKVersion
+
+jobs:
+  public-api-check:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+            fetch-depth: 0
+
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 3.0
+
+      - name: Check for new public APIs
+        id: check-api
+        run: |
+          # Run the Ruby script and make a list of new public APIs 
+          echo "Checking for new Public APIs..."
+          OUTPUT=$(ruby ci_scripts/check_for_new_public_apis.rb ${{ github.workspace }})
+          echo "${OUTPUT}"
+          echo "API_CHECK_OUTPUT<<EOF" >> $GITHUB_ENV
+          echo "${OUTPUT}" >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
```

Comment:
I guess the only way this could break is if ${OUTPUT} contained an EOF marker, which I think is impossible?

Reply:
++, I think it's impossible too. We can keep an eye on it though.

Reply:
++, I think it's impossible too. We can keep an eye on it though.